automation/actions/cacheLPTTotalSupply/index.ts

```diff
@@ -0,0 +1,40 @@
+import {ActionFn, Context, Event} from '@tenderly/actions';
+import {ethers} from 'ethers';
+import {waitForTx, waitToRelayTxsToL2} from '../../../utils/arbitrum';
+import {getArbParams, getCache} from './helpers';
+import {ARBITRUM_NETWORK} from '../../../deploy/constants';
+
+export const triggerCache: ActionFn = async (
+    context: Context,
+    _event: Event,
+) => {
+  const infuraKeyL1 = await context.secrets.get('INFURA_KEY_ETH');
+  const infuraKeyL2 = await context.secrets.get('INFURA_KEY_ARB');
+
+  const ethProvider = new ethers.providers.InfuraProvider(infuraKeyL1);
+  const arbProvider = new ethers.providers.InfuraProvider(infuraKeyL2);
+
+  const pvtKey = await context.secrets.get('PVT_KEY_SIGNER');
+  const signer = new ethers.Wallet(pvtKey);
+
+  const cache = getCache(ethProvider);
+
+  const {maxGas, gasPriceBid, maxSubmissionCost, ethValue} =
+    await getArbParams(cache, arbProvider, signer.address);
+
+  const tx = cache
+      .connect(signer)
+      .cacheTotalSupply(maxGas, gasPriceBid, maxSubmissionCost, {
+        value: ethValue,
+      });
+
+  const {l1TxHash, l2TxHash} = await waitToRelayTxsToL2(
+      waitForTx(tx),
+      ARBITRUM_NETWORK.rinkeby.inbox,
```

Comment:
I assume that it's rinkeby only momentarily for testing. Can we make it more flexible as to support both testnet and mainnet (i.e.: depending on some environment variable/input parameter)? As a side-note, are you familiar with the hardhat-tenderly plugin [1]? It might make the integration of tenderly actions in the codebase more seamless (however it seems it has not not been updated in a while so I'm not sure whether it supports tenderly actions since they seem to be a relatively recent feature)

[1] https://github.com/Tenderly/hardhat-tenderly

Reply:
unfortunately the plugin does not support actions yet. I don't think there is a way to make it flexible other than deploying a fresh action targetting testnet. changed it to target mainnet in 2744b82